src/main/java/org/jenkinsci/plugins/workflow/log/DelayBufferedOutputStream.java

```diff
@@ -54,6 +58,45 @@ private Tuning() {}
         static final Tuning DEFAULT = new Tuning();
     }
 
+    /**
+     * The interesting state of the buffered stream, kept as a separate object so that {@link FlushRef} can hold on to it.
+     */
+    private static final class Buffer {
+
+        final OutputStream out;
+        private final byte[] dat;
+        private int pos;
```

Comment:
I feel like we're tempting fate here by having both unsynchronized and synchronized access to these values without making them `volatile`

Reply:
All access is synchronized.

Reply:
At the *top* level -- it just seems silly to mix these around internally.

Reply:
At the top level, yes.  Internally mixing the two forms feels a bit nonsensical.  

Anyway, this is pretty much the same comment Devin made, just phrased differently.